Core/Core/Inbox/InboxCoursePicker/View/InboxCoursePickerView.swift

```diff
@@ -0,0 +1,174 @@
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public struct InboxCoursePickerView: View {
+    @ObservedObject private var viewModel: InboxCoursePickerViewModel
+
+    init(viewModel: InboxCoursePickerViewModel) {
+        self.viewModel = viewModel
+    }
+
+    public var body: some View {
+        content
+            .navigationTitleStyled(Text("Select Course", bundle: .core).font(.semibold17).foregroundColor(.textDarkest))
+            .navigationBarTitleDisplayMode(.inline)
+    }
+
+    @ViewBuilder
+    private var content: some View {
+        ScrollView {
+            switch viewModel.state {
+            case .loading:
+                ProgressView()
+                    .progressViewStyle(.indeterminateCircle())
+                    .padding(12)
+            case .data:
+                VStack(spacing: 0) {
+                    courses(courses: viewModel.courses)
+                    groups(groups: viewModel.groups)
+                }
+            case .empty, .error:
+                Text("Some error occured", bundle: .core)
+            }
+        }
+    }
+
+    private func error(message: String) -> some View {
+        Text(message)
+            .font(.regular17)
+            .foregroundColor(.textDarkest)
+    }
+
+    private var separator: some View {
+        Color.borderMedium
+            .frame(height: 0.5)
+    }
+
+    private func courses(courses: [Course]) -> some View {
+        VStack(spacing: 0) {
+            headerView(NSLocalizedString("Courses", bundle: .core, comment: ""))
+                .accessibilitySortPriority(5)
+            ForEach(courses, id: \.id) { course in
+                courseRow(course)
+            }
+        }
+    }
+
+    private func groups(groups: [Group]) -> some View {
+        VStack(spacing: 0) {
+            headerView(NSLocalizedString("Groups", bundle: .core, comment: ""))
+            ForEach(groups, id: \.id) { group in
+                groupRow(group)
+            }
+        }
+    }
+
+    private func headerView(_ header: String) -> some View {
+        VStack(spacing: 0) {
+            separator
+            Text(header)
+                .font(.regular14)
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .foregroundStyle(Color.textDark)
+                .padding(.horizontal, 12)
+                .padding(.vertical, 16)
+                .background(Color.backgroundLight)
```

Comment:
Marking this text as header for voiceover would make the purpose of this text more obvious.